### PR TITLE
Added {Get|Set}AbilityIncreaseByLevel()

### LINKED ISSUE #1671: [Feature Request] Adding NWNX_Creature_GetAbilityPointsByLevel()
We currently have a way to get the Skill ranks and feats of a creature by level. It would be great to also be able to do this with Abilities.

### PR DESCRIPTION
**Pull request by Cjreek**
closes #1671

### CODE REVIEW

**Comment by Cjreek** [comment]
**Please don't merge yet!**: The ELC plugin uses Ability::MAX, which has been changed and now includes Ability::None. This will most likely break the related code. We'll have to talk about what to do here. 

Either the ELC code needs to be changed or I have to loop from 0-6 manually without the ability constants. Or something else I'm not thinking about right now.

**Comment by Daztek** [comment]
Maybe just change all `<= Ability::MAX` references to `< Ability::MAX`

**Comment by Cjreek** [comment]
Okay it's fixed

File: Plugins/Creature/NWScript/nwnx_creature.nss [hunk 1]
```diff
@@ -61,6 +61,8 @@ const int NWNX_CREATURE_PROJECTILE_VFX_SONIC        = 5;
 const int NWNX_CREATURE_PROJECTILE_VFX_RANDOM       = 6; ///< Random Elemental VFX
 /// @}
 
+const int ABILITY_NONE = 6;
```
Comment (Daztek) [inline Plugins/Creature/NWScript/nwnx_creature.nss:64]: Would prefer this to be prefixed with `NWNX_CREATURE_` + a note in the function desc.

File: CHANGELOG.md [hunk 2]
```diff
@@ -11,13 +11,15 @@ https://github.com/nwnxee/unified/compare/build8193.35.40...HEAD
 - Feat: added modifier `NWNX_FEAT_MODIFIER_SPELLSAVEDCFORSCHOOL` to modify a creature's spell DC for a spell school
 - Feat: added modifier `NWNX_FEAT_MODIFIER_SPELLSAVEDCFORSPELL` to modify a creature's spell DC for an individual spell
 - Events: added event `NWNX_ON_SPELL_FAILED_{BEFORE|AFTER}` which fires when the casting of a spell did not finish for any reason.
+- Creature: added constant `ABILITY_NONE`
```
Comment (Daztek) [inline CHANGELOG.md:14]: No real need for this I think

**Comment by Cjreek** [comment]
Updated the constant, function documentations and changelog

**Pull request merged**